nx/lib/nx/defn/compiler.ex

```diff
@@ -421,6 +474,18 @@ defmodule Nx.Defn.Compiler do
     {{call, meta, [ast, fun]}, state}
   end
 
+  defp normalize({{:., _, [Nx.Defn.Assertions, _function]} = call, meta, [ast, fun]}, state) do
```

Comment:
This should probably not be here, just added to get the tests to pass/code to compile without warnings.

I think this is related to what we talked about over remote calls. Basically, I don't want to normalize args passed to a transform in this case. This came up when replacing the `transform` in `assert_shape_pattern`

Reply:
(this is basically a copy-paste from the :transform clause above, by the way)

Reply:
Yeah... the only way I can think to address this is to introduce some sort of `match_shape` macro. Then we can also introduce `match_type` (which will help clean up adjoint/transform). `match_shape` is easy... so perhaps we can start with it. But `match_type` also has complications because often we just want to ask if something is a float and have it work for both f/bf. Similar for integers.

Anyway, let's add `match_shape`?

Reply:
You can remove this now. :)